### PR TITLE
Followups for getdeploymentinfo

### PR DESCRIPTION
**Pull request by ajtowns**
Documentation, comments and trivial code changes to followup #23508.

### CODE REVIEW

File: doc/release-notes-23508.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+------------
+
+- Information on soft fork status has been moved from `getblockchaininfo`
+  to the new `getdeploymentinfo` RPC which allows querying soft fork status at any
+  block, rather than just at the chain tip. Inclusion of soft fork
+  status in `getblockchaininfo` can currently be restored using the
+  configuration `-deprecatedrpc=softforks`, but this will be removed in
+  a future release. Note that in either case, the `status` field
+  now reflects the status of the current block rather than the next
+  block. (#23508)
```
Comment (MarcoFalke) [inline doc/release-notes-23508.md:11]: could move to the main file to avoid having to do it later? (This should be the last snippet for the next release)

**Comment by MarcoFalke** [comment]
Maybe also https://github.com/bitcoin/bitcoin/pull/23508#discussion_r785158521 ?

**Comment by ajtowns** [comment]
Was leaving the `SoftForkPushBackDesc` argument name change 'til other work that would also touch that, but meh, added.

**Review by maflcko — APPROVED** [review]
LGTM

**Comment by luke-jr** [comment]
We haven't used the variable style behind `pindex` for years now. Why rename to it?

**Comment by Sjors** [comment]
> We haven't used the variable style behind `pindex` for years now. Why rename to it?

We can name something else (as long as it's clear we're not necessarily dealing with the tip).

**Comment by ajtowns** [comment]
Changed to blockindex.

**Comment by JeremyRubin** [comment]
shrug about renaming _tip_, easy enough to fix the conflict either way.

**Comment by Sjors** [comment]
@JeremyRubin a rebase beats an off by one soft fork activation error due to unreadable activation-related code :-)

**Pull request merged**